framework/Models/PostProcessors/validationAlgorithms/Probabilistic.py

```diff
@@ -135,22 +71,121 @@ def _handleInput(self, paramInput):
     """
     super()._handleInput(paramInput)
 
-  def run(self, datasets, **kwargs):
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action. In this case it loads the
+      results to specified dataObject
+      @ In, inputIn, list, dictionary of data to process
+      @ Out, outputDict, dict, dictionary containing the post-processed results
+    """
+    # inpVars, outVars, dataSet = inputIn['Data'][0]
+    dataSets = [data for _, _, data in inputIn['Data']]
```

Comment:
I suggest to store the dataSets into a dictionary, i.e.

dataDict = {data.attrs['name']: data for _, _, data in inputIn['Data']}

In this case you can get ride of _evaluate method, and much easier to retrieve the feat/targ data from the dataDict.